frame/scheduler/src/lib.rs

```diff
@@ -496,10 +502,30 @@ impl<T: Trait> Module<T> {
 			Self::deposit_event(RawEvent::Canceled(when, index));
 			Ok(())
 		} else {
-			Err(Error::<T>::FailedToCancel)?
+			Err(Error::<T>::NotFound)?
 		}
 	}
 
+	fn do_reschedule(
+		(when, index): TaskAddress<T::BlockNumber>,
+		new_time: DispatchTime<T::BlockNumber>,
+	) -> Result<TaskAddress<T::BlockNumber>, DispatchError> {
+		let new_time = Self::resolve_time(new_time)?;
+
+		Agenda::<T>::try_mutate(when, |agenda| -> DispatchResult {
+			let task = agenda.get_mut(index as usize).ok_or(Error::<T>::NotFound)?;
+			let task = task.take().ok_or(Error::<T>::NotFound)?;
+			Agenda::<T>::append(new_time, Some(task));
+			Ok(())
+		})?;
```

Comment:
I'm surprised this is working because `Agenda::<T>::append(new_time, Some(task));` is done inside try_mutate, and try_mutate should at the end write the mutated agend into storage thus append operation should be replaced no ?

Reply:
Good finding. I added check to return error when reschedule to same block.